### PR TITLE
ERMS-1989 + ERMS-2643

### PR DESCRIPTION
**Pull request by Moetez16**

### CODE REVIEW

File: app/grails-app/controllers/de/laser/AddressController.groovy [hunk 1]
```diff
@@ -31,9 +31,9 @@ class AddressController extends AbstractDebugController {
 				break
 			case 'POST':
                 if (formService.validateToken(params)) {
-
+                    println("Test")
```
Comment (klodav) [inline app/grails-app/controllers/de/laser/AddressController.groovy:34]: Test successful!

File: app/grails-app/controllers/de/laser/AddressController.groovy [hunk 2]
```diff
@@ -42,6 +42,7 @@ class AddressController extends AbstractDebugController {
 
                     flash.message = message(code: 'default.created.message', args: [message(code: 'address.label'), addressInstance.name])
                 }
+                println("Mow")
```
Comment (klodav) [inline app/grails-app/controllers/de/laser/AddressController.groovy:45]: Whoof!

File: app/grails-app/views/templates/cpa/_person_table.gsp [hunk 3]
```diff
@@ -6,123 +6,161 @@
...
+                            <g:if test="${controllerName == 'myInstitution' && actionName == 'addressbook'}">
+                                <div class="la-flexbox">
+                                    <i class="icon university la-list-icon"></i>
+                                    <g:link controller="organisation" action="addressbook"
+                                            id="${role.org?.id}">${role.org}</g:link>
+                                </div>
+                            </g:if>
                             <g:if test="${role.functionType}">
-                                (${role.functionType?.getI10n('value')}) <br />
+                                ${role.functionType?.getI10n('value')}
```
Comment (klodav) [inline app/grails-app/views/templates/cpa/_person_table.gsp:84]: ? not necessary here

File: app/grails-app/views/templates/cpa/_person_table.gsp [hunk 4]
```diff
@@ -6,123 +6,161 @@
...
                             </g:if>
                             <g:if test="${role.positionType}">
                                 (${role.positionType?.getI10n('value')})
```
Comment (klodav) [inline app/grails-app/views/templates/cpa/_person_table.gsp:87]: ? not necessary here

File: app/grails-app/views/myInstitution/myPublicContacts.gsp [hunk 5]
```diff
@@ -2,171 +2,166 @@
...
+        </g:else>
+            <semui:actionsDropdownItem notActive="true" data-semui="modal" href="#copyFilteredEmailAddresses_ajaxModal"
+                                       message="menu.institutions.copy_emailaddresses.button"/>
+        </semui:actionsDropdown>
+    </semui:controlButtons>
+
+
+    <g:render template="/person/formModal" model="['org'               : institution,
+                                                   'isPublic'          : RDStore.YN_YES,
+                                                   'presetFunctionType': RefdataValue.getByValueAndCategory('General contact person', RDConstants.PERSON_FUNCTION)
```
Comment (klodav) [inline app/grails-app/views/myInstitution/myPublicContacts.gsp:62]: RDStore.PRS_FUNC_GENERAL_CONTACT_PRS

File: app/grails-app/views/myInstitution/myPublicContacts.gsp [hunk 6]
```diff
@@ -2,171 +2,166 @@
...
-                        next="${message(code: 'default.paginate.next')}"
-                        prev="${message(code: 'default.paginate.prev')}"
-                        max="${max}"
-                        total="${num_visiblePersons}"/>    </div>
+            <div class="field la-field-right-aligned">
+                <label></label>
+                <a href="${request.forwardURI}"
+                   class="ui reset primary button">${message(code: 'default.button.reset.label')}</a>
+                <input type="submit" class="ui secondary button"
+                       value="${message(code: 'default.button.filter.label', default: 'Filter')}">
```
Comment (klodav) [inline app/grails-app/views/myInstitution/myPublicContacts.gsp:109]: please remove default fallback for i18n message

File: app/grails-app/views/organisation/show.gsp [hunk 7]
```diff
@@ -442,35 +446,37 @@
...
-                                           %{--data-href="#contactFormModal"/>--}%
-                                    %{--<g:render template="/contact/formModal" model="['orgId': orgInstance?.id]"/>--}%
-                                %{--</g:if>--}%
-                            %{--</dd>--}%
-                        %{--</dl>--}%
                         <dl>
+                            <dt><g:message code="org.contacts.label" /></dt>
+                            <dd>
+                                <div class="ui divided middle aligned selection list la-flex-list">
+                                    <g:each in="${orgInstance?.contacts?.toSorted()}" var="c">
```
Comment (klodav) [inline app/grails-app/views/organisation/show.gsp:453]: orgInstance?

**Review by klodav — COMMENTED** [review]
minor remarks

**Comment by Moetez16** [comment]
Done!

**Review by klodav — APPROVED** [review]

**Pull request merged**